File: src/test/java/com/leesh/inflpick/mock/FakeInfluencerRepository.java
package com.leesh.inflpick.mock;

import com.leesh.inflpick.common.port.PageDetails;
import com.leesh.inflpick.common.port.PageQuery;
import com.leesh.inflpick.influencer.adapter.out.persistence.mongo.InfluencerDocument;
import com.leesh.inflpick.influencer.core.domain.Influencer;
import com.leesh.inflpick.influencer.port.InfluencerSortType;
import com.leesh.inflpick.influencer.port.out.InfluencerNotFoundException;
import com.leesh.inflpick.influencer.port.out.InfluencerRepository;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.List;

public class FakeInfluencerRepository implements InfluencerRepository {

    private final List<Influencer> data = Collections.synchronizedList(new ArrayList<>());

    @Override
    public String save(@NotNull Influencer influencer) {
        InfluencerDocument document = InfluencerDocument.from(influencer);
        if (data.stream().anyMatch(d -> d.getId().equals(document.getId()))) {
            data.removeIf(d -> d.getId().equals(document.getId()));
            data.add(influencer);
        } else {
            data.add(influencer);
        }
        return influencer.getId();
    }

    @Override
    public long count() {
        return data.size();
    }

    @Override
    public @NotNull Influencer getById(String uuid) throws InfluencerNotFoundException {
        return data.stream()
                .filter(d -> d.getId().equals(uuid))
                .findFirst()
                .orElseThrow(() -> new InfluencerNotFoundException(uuid));
    }

    @Override
    public void deleteById(String id) {
        data.removeIf(d -> d.getId().equals(id));
    }

    @Override
    public PageDetails<Collection<Influencer>> getPage(PageQuery<InfluencerSortType> query) {
        return null;
    }

}

File: src/test/kotlin/com/leesh/inflpick/mock/FakeInfluencerRepository.kt
package com.leesh.inflpick.mock

import com.leesh.inflpick.influencer.domin.InfluencerV2
import com.leesh.inflpick.influencer.service.port.InfluencerRepositoryV2
import java.util.concurrent.CopyOnWriteArrayList

class FakeInfluencerRepository(data: List<InfluencerV2>) : InfluencerRepositoryV2 {

    private val data = CopyOnWriteArrayList<InfluencerV2>()

    init {
        this.saveAll(data)
    }

    override fun saveAll(influencerV2s: List<InfluencerV2>): List<InfluencerV2> {
        return influencerV2s.map { save(it) }
    }

    override fun save(influencerV2: InfluencerV2): InfluencerV2 {
        data.removeIf { it == influencerV2 }
        val updatedInfluencer = influencerV2.copy(id = FakeDomainId())
        data.add(updatedInfluencer)
        return influencerV2
    }

    override fun searchByName(keyword: String): List<InfluencerV2> {
        return data.filter { it.name.contains(keyword) }
    }

}